### PR TITLE
Limit Sql body size

### LINKED ISSUE #5596: Limit Sql size
Please answer these questions before submitting your issue.

- Why do you submit this issue?
- [ ] Question or discussion
- [ ] Bug
- [ ] Requirement
- [x] Feature or performance improvement

___
### Requirement or improvement
- Please describe your requirements or improvement suggestions.
The bigger sql (refer #5479 ) will occupy too much memory, even more will cause oom issue. we should set a threshold to  limit the sql size.

### PR DESCRIPTION
**Pull request by xbkaishui**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] Bug fix
- [ ] New feature provided
- [ ] Improve performance

- Related issues
resolved #5596 
___
### Bug fix
- Bug description.

- How to fix?
set a thredhold to limit sql body size
___
### New feature or improvement
- Describe the details and related test reports.

### CODE REVIEW

File: apm-sniffer/apm-sdk-plugin/jdbc-commons/src/main/java/org/apache/skywalking/apm/plugin/jdbc/JDBCPluginConfig.java [hunk 1]
```diff
@@ -53,6 +60,13 @@
...
+            /**
+             * For the sake of performance, SkyWalking won't save the entire sql body into the tag, but only the first
+             * {@code SQL_BODY_MAX_LENGTH} characters.
+             * <p>
+             * Set a negative number to save the complete sql body to the tag.
+             */
+            public static int SQL_BODY_MAX_LENGTH = 2048;
         }
 
         public static class MARIADB {
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/jdbc-commons/src/main/java/org/apache/skywalking/apm/plugin/jdbc/JDBCPluginConfig.java:72]: I think we missed an annotation for this, this config would never be initialized. :)

Also, MySQL/POSTGRESQL/MARIADB seems the same, is there any reason to separate them? @ascrutae. I think the only case is, an app is using 2 kinds of DB, but wants to trace one but not for the other. Is this really reasonable?

File: apm-sniffer/apm-sdk-plugin/jdbc-commons/src/main/java/org/apache/skywalking/apm/plugin/jdbc/SqlBodyBuilder.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.jdbc;
+
+public class SqlBodyBuilder {
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/jdbc-commons/src/main/java/org/apache/skywalking/apm/plugin/jdbc/SqlBodyBuilder.java:21]: If MySQL/POSTGRESQL/MARIADB configs are going to be merged, then we don't need to initial this class every time.

File: docs/en/setup/service-agent/java-agent/README.md [hunk 3]
```diff
@@ -123,10 +123,13 @@ property key | Description | Default |
...
 `plugin.postgresql.trace_sql_parameters`|If set to true, the parameters of the sql (typically `java.sql.PreparedStatement`) would be collected.|`false`|
 `plugin.postgresql.sql_parameters_max_length`|If set to positive number, the `db.sql.parameters` would be truncated to this length, otherwise it would be completely saved, which may cause performance problem.|`512`|
+`plugin.postgresql.sql_body_max_length`|If set to positive number, the `db.statement` would be truncated to this length, otherwise it would be completely saved, which may cause performance problem.|`2048`|
 `plugin.mariadb.trace_sql_parameters`|If set to true, the parameters of the sql (typically `java.sql.PreparedStatement`) would be collected.|`false`|
 `plugin.mariadb.sql_parameters_max_length`|If set to positive number, the `db.sql.parameters` would be truncated to this length, otherwise it would be completely saved, which may cause performance problem.|`512`|
+`plugin.mariadb.sql_body_max_length`|If set to positive number, the `db.statement` would be truncated to this length, otherwise it would be completely saved, which may cause performance problem.|`2048`|
```
Comment (wu-sheng) [inline docs/en/setup/service-agent/java-agent/README.md:132]: Config doc relates to the conclusion about the `MySQL/POSTGRESQL/MARIADB` config.

File: apm-sniffer/apm-sdk-plugin/jdbc-commons/src/main/java/org/apache/skywalking/apm/plugin/jdbc/JDBCPluginConfig.java (hunk 1 above)
Comment (xbkaishui) [reply apm-sniffer/apm-sdk-plugin/jdbc-commons/src/main/java/org/apache/skywalking/apm/plugin/jdbc/JDBCPluginConfig.java:72]: Yes,  you found a :bug: , So can I uniform  MySQL/POSTGRESQL/MARIADB  config to a single one ? 
for blew config 

- TRACE_SQL_PARAMETERS
- SQL_PARAMETERS_MAX_LENGTH
- SQL_BODY_MAX_LENGTH

Comment (wu-sheng) [reply apm-sniffer/apm-sdk-plugin/jdbc-commons/src/main/java/org/apache/skywalking/apm/plugin/jdbc/JDBCPluginConfig.java:72]: I prefer so, waiting for @ascrutae 's idea.

Comment (ascrutae) [reply apm-sniffer/apm-sdk-plugin/jdbc-commons/src/main/java/org/apache/skywalking/apm/plugin/jdbc/JDBCPluginConfig.java:72]: Me too, +1

Comment (wu-sheng) [reply apm-sniffer/apm-sdk-plugin/jdbc-commons/src/main/java/org/apache/skywalking/apm/plugin/jdbc/JDBCPluginConfig.java:72]: @xbkaishui Seems these configs could be merged.

Comment (xbkaishui) [reply apm-sniffer/apm-sdk-plugin/jdbc-commons/src/main/java/org/apache/skywalking/apm/plugin/jdbc/JDBCPluginConfig.java:72]: Ok, will change

Comment (xbkaishui) [reply apm-sniffer/apm-sdk-plugin/jdbc-commons/src/main/java/org/apache/skywalking/apm/plugin/jdbc/JDBCPluginConfig.java]: done

File: apm-sniffer/apm-sdk-plugin/jdbc-commons/src/main/java/org/apache/skywalking/apm/plugin/jdbc/SqlBodyBuilder.java (hunk 2 above)
Comment (xbkaishui) [reply apm-sniffer/apm-sdk-plugin/jdbc-commons/src/main/java/org/apache/skywalking/apm/plugin/jdbc/SqlBodyBuilder.java]: done

File: docs/en/setup/service-agent/java-agent/README.md (hunk 3 above)
Comment (xbkaishui) [reply docs/en/setup/service-agent/java-agent/README.md]: done

**Review by wu-sheng — COMMENTED** [review]
Have you changed the plugin tests? There are expected YAML data files there. As you change the config, you actually break the old config keys.

**Comment by xbkaishui** [comment]
No, let me check one by one

**Comment by xbkaishui** [comment]
Hi @wu-sheng , fixed e2e test scenario, those 2 fails are caused by maven issue.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**